2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/util.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2013-2017 Snowplow Analytics Ltd. All rights reserved.
+ *
+ * This program is licensed to you under the Apache License Version 2.0, and
+ * you may not use this file except in compliance with the Apache License
+ * Version 2.0.  You may obtain a copy of the Apache License Version 2.0 at
+ * http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the Apache License Version 2.0 is distributed on an "AS
+ * IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the Apache License Version 2.0 for the specific language
+ * governing permissions and limitations there under.
+ */
+package com.snowplowanalytics.snowplow.collectors.scalastream
+
+import scala.util.{Failure, Try, Success}
+
+import scalaz._
+import Scalaz._
+
+object util {
+  // to rm once 2.12 as well as the right projections
+  def fold[A, B](t: Try[A])(ft: Throwable => B, fa: A => B): B = t match {
```

Comment:
`ft: Throwable => B` looks a bit weird to me. Is the purpose of the provided function to use a throwable and return just B?

Reply:
that's just how [fold](https://www.scala-lang.org/api/2.12.4/scala/util/Try.html#fold[U](fa:Throwable=%3EU,fb:T=%3EU):U) works